hail/python/hailtop/auth/tokens.py

```diff
@@ -8,6 +9,14 @@
 log = logging.getLogger('gear')
 
 
+def session_id_encode_to_str(session_id_bytes):
+    return base64.urlsafe_b64encode(session_id_bytes).decode('ascii')
+
+
+def session_id_decode_from_str(session_id_str):
+    return base64.urlsafe_b64decode(session_id_str.encode('ascii'))
```

Comment:
Should this be base64.urlsafe_b64decode(session_id_str).encode('ascii')?

Reply:
I added types that should make this a bit more clear. The input `session_id_str` is a Python `str`. I "encode" that string as a Python `bytearray` using the ASCII encoding. Then I send that into the base64 decoder which converts a bytearray of base64 encoded ASCII into a byte array of regular old ASCII.


Step-wise:
`encode_to_str`
- bytes (`bytearray`)
- base64-encoded bytes (`bytearray`)
- base64-encoded ASCII string (`str`)

`decode_from_str`
- base64-encoded ASCII string (`str`)
- base64-encoded bytes (`bytearray`)
- bytes (`bytearray`)

It's maybe more clear with intermediate variables:
```python3
    session_id_base64_bytes = session_id_str.encode('ascii')
    session_id_just_bytes = base64.urlsafe_b64decode(session_id)
    return session_id_just_bytes
```